internal/command/machine/run.go

```diff
@@ -588,10 +588,24 @@ func determineMachineConfig(ctx context.Context, initialMachineConf api.MachineC
 
 	if guestSize := flag.GetString(ctx, "size"); guestSize != "" {
 		guest, ok := api.MachinePresets[guestSize]
+
 		if !ok {
+			var machine_type string
+
+			if strings.HasPrefix(guestSize, "shared") {
+				machine_type = "shared"
+
+			} else if strings.HasPrefix(guestSize, "performance") {
+				machine_type = "performance"
+
+			} else {
+				return machineConf, fmt.Errorf("invalid machine preset requested, '%s', expected 'shared' or 'performance'", guestSize)
```

Comment:
This error is slightly misleading, we expected it to _start with_ 'shared' or 'performance' right?

Reply:
You're right, I'll fix that now